### PR TITLE
Add test result detail view

### LINKED ISSUE #490: Add details view to test results page
## TODOS
Implement v1 of details view (done)
- [x] Add dropdown to Test Results table (also needed in https://github.com/CDCgov/prime-simplereport/issues/426)
- [x] add "Mark as Error" link with an alert to confirm
- [x] add reason for error
- [x] cross off the test result if there is an error

Implement v2 of details view (see screenshots below or [engineering prototype](https://www.figma.com/proto/CEzBM8kUw8LyWdfAhUThYb/Engineering-Ready-Prototype?node-id=23%3A260&scaling=scale-down&page-id=0%3A1))
- [x] Add a “View details” option to actions menu
- [x] Add details view UI 
- [x] In the details view, if the result is marked an error, show all text as grey with a strike through

## Mocks/Specs: 
https://docs.google.com/document/d/1qqv4kH0sb865NYdk6rpnU9cQMWlu6mFHK2J8CsQOmOA/edit#

Add "View details" option to action menu

Add details view UI

Details UI if result has been marked an error

## Ref
EPIC: https://github.com/CDCgov/prime-simplereport/issues/473

### PR DESCRIPTION
**Pull request by nickscialli-usds**
## Related Issue or Background Info

- Resolves #490 

## Changes Proposed

- Adds new modal view
- Changes `constants.js` -> `constants.ts`
- Adds symptom parser util and tests it

## Screen Recording

view details

### CODE REVIEW

**Comment by nickscialli-usds** [comment]
Just noticed I need to add the time (e.g., 9:00am) to the "Test date" row, will make that change now

**Comment by katiealoisi** [comment]
Thx @nickscialli-usds! Can you also make the following capitalization fixes?

Change "Test Details" to "Test details" 
Change "Test Date" to "Test date"
Change "SARS-CoV-2 Result" to "SARS-CoV-2 result"
Change "Symptom Onset" to "Symptom onset"

**Comment by nickscialli-usds** [comment]
@katiealoisi Will do!

**Comment by nickscialli-usds** [comment]
@katiealoisi here's a screenshot of the updates

test-details

**Comment by katiealoisi** [comment]
Looks great!

File: frontend/src/patientApp/timeOfTest/constants.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+// BEGIN things that should be service calls
```
Comment (nullflux) [inline frontend/src/patientApp/timeOfTest/constants.ts:1]: Thanks for this!

File: frontend/src/app/testResults/TestResultsList.tsx [hunk 2]
```diff
@@ -250,10 +257,24 @@ export const DetachedTestResultsList: any = ({
...
+  );
 
   return (
     <main className="prime-home">
+      {detailsModalId && (
+        <TestResultDetailsModal
+          testResultId={detailsModalId}
+          closeModal={() => {
+            setDetailsModalId(undefined);
+            refetch();
```
Comment (nullflux) [inline frontend/src/app/testResults/TestResultsList.tsx:274]: Just curious why we're refetching here?

**Review by nullflux — APPROVED** [review]
LGTM! One non-blocking question

File: frontend/src/app/testResults/TestResultsList.tsx (hunk 2 above)
Comment (nickscialli-usds) [reply frontend/src/app/testResults/TestResultsList.tsx:274]: This is copypasta from the print modal. I can (and probably should) remove it. I wonder if the print modal (erroneously) copied it from something that actually edited data and therefore would require a refetch.

**Pull request merged**